test-framework/core/src/main/java/org/keycloak/testframework/https/ManagedCertificate.java

```diff
@@ -0,0 +1,110 @@
+package org.keycloak.testframework.https;
+
+import org.jboss.logging.Logger;
+import org.keycloak.common.crypto.CryptoProvider;
+import org.keycloak.common.util.KeystoreUtil;
+
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.KeyPair;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+
+public class ManagedCertificate {
```

Comment:
I'd rename this to certificates, as we probably want to expand on this in the future to also support generating client certificates for mTLS use-cases
```suggestion
public class ManagedCertificates {
```